src/decirest.erl

```diff
@@ -268,6 +270,28 @@ change_module(Module, #{mro := MRO} = State) ->
   [{Handler, _OldModule} | Tail] = lists:reverse(MRO),
   State#{module => Module, main_module => Module, mro => lists:reverse([{Handler, Module}| Tail])}.
 
+
+-spec get_handler(State) -> Handler when
+  State :: map(),  % Decirest state
+  Handler :: decirest_single_handler | decirest_collection_handler.
+get_handler(#{module := Module} = State) ->
+  get_handler(Module, State).
+
+
+-spec get_handler(Module, State) -> Handler when
+  Module :: atom(),
+  State :: map(),
+  Handler :: decirest_single_handler | decirest_collection_handler | module_dont_exist_in_state.
+get_handler(Module, State) ->
+  #{mro := MRO} = State,
+  case lists:keyfind(Module, 2, MRO) of
+    false ->
```

Comment:
I think we can skip the case. Module should always be in mro.

Reply:
get_hander/2 allows to ask for any module, and so potentially could ask for something not in State

Reply:
Not sure when that would be used. Only last can be collection or single right? If we want to keep it we should tag them with ok/error.